src/uct/ib/ud/base/ud_iface.c

```diff
@@ -30,214 +30,124 @@ static ucs_stats_class_t uct_ud_iface_stats_class = {
 };
 #endif
 
-/* cppcheck-suppress ctunullpointer */
-SGLIB_DEFINE_LIST_FUNCTIONS(uct_ud_iface_peer_t, uct_ud_iface_peer_cmp, next)
-SGLIB_DEFINE_HASHED_CONTAINER_FUNCTIONS(uct_ud_iface_peer_t,
-                                        UCT_UD_HASH_SIZE,
-                                        uct_ud_iface_peer_hash)
 
 static void uct_ud_iface_free_pending_rx(uct_ud_iface_t *iface);
 static void uct_ud_iface_free_async_comps(uct_ud_iface_t *iface);
 
-
-void uct_ud_iface_cep_init(uct_ud_iface_t *iface)
+void uct_ud_iface_cep_init(uct_ud_iface_t *iface, size_t address_length,
+                           const ucs_conn_match_ops_t *ops)
 {
-    sglib_hashed_uct_ud_iface_peer_t_init(iface->peers);
+    ucs_conn_match_init(&iface->conn_match_ctx, address_length, ops);
 }
 
-static void
-uct_ud_iface_cep_cleanup_eps(uct_ud_iface_t *iface, uct_ud_iface_peer_t *peer)
+void uct_ud_iface_cep_cleanup(uct_ud_iface_t *iface)
```

Comment:
maybe remove this wrapper function and call ucs_conn_match firectly?
also for uct_ud_iface_cep_init